src/protocols/multi_party_ecdsa/gg_2020/orchestrate.rs

```diff
@@ -0,0 +1,1241 @@
+        if var_os("WRITE_FILE").is_some() {
+            use std::fs::OpenOptions;
+            let mut json_file = OpenOptions::new()
+                .append(true)
+                .open(&format!("{}.txt", $op))
+                .unwrap();
+            let index = $index;
+            let stage = $stage;
+            let op = $op;
+            let json = $json;
+            json_file
+                .write_all(format!("Output {} stage {} index {}\n", op, stage, index).as_bytes())
+                .unwrap();
+            json_file
+                .write_all(format!("{}\n", json).as_bytes())
+                .unwrap();
+        }
+    }};
+}
+
+// The Distributed key generation protocol can work with a broadcast channel.
+// All the messages are exchanged p2p.
+// On the contrary, the key generation process can be orchestrated as below.
+// All the participants do some work on each stage and return some data.
+// This data needs to be filtered/collated and sent back as an input to the next stage.
+// This test helper is just a demonstration of the same.
+//
+pub fn keygen_orchestrator(params: Parameters) -> Result<KeyPairResult, ErrorType> {
+    let op = "keygen".to_string();
+    if var_os("WRITE_FILE").is_some() {
+        File::create(&format!("{}.txt", &op)).unwrap();
+    }
+    //
+    // Values to be kept private(Each value needs to be encrypted with a key only known to that
+    // party):
+    // 1. party_keys.u_i
+    // 2. party_keys.dk
+    let mut party_keys_vec_l = vec![];
+    // Nothing private in the commitment values.
+    let mut bc1_vec_l = vec![];
+    // Values to be kept private(Each value needs to be encrypted with a key only known to that
+    // party):
+    // 1. decommitment values in KeyGenDecommitMessage1 need to be encrypted until they are sent
+    let mut decom_vec_l = vec![];
+    // Nothing private in the below vector.
+    let mut h1_h2_N_tilde_vec_l = vec![];
+    for i in 0..params.share_count {
+        let input = KeyGenStage1Input { index: i as usize };
+        write_input!(i, 1, &op, serde_json::to_string_pretty(&input).unwrap());
+        let res = keygen_stage1(&input);
+        write_output!(i, 1, &op, serde_json::to_string_pretty(&res).unwrap());
+        party_keys_vec_l.push(res.party_keys_l);
+        bc1_vec_l.push(res.bc_com1_l);
+        decom_vec_l.push(res.decom1_l);
+        h1_h2_N_tilde_vec_l.push(res.h1_h2_N_tilde_l);
+    }
+    let mut vss_scheme_vec_l = vec![];
+    // Values to be kept private(Each value needs to be encrypted with a key only known to that
+    // party):
+    // 1. secret_shares_vec_l[i] -
+    //    party.
+    let mut secret_shares_vec_l = vec![];
+    let mut index_vec = vec![];
+    for i in 0..params.share_count {
+        let input = KeyGenStage2Input {
+            index: i as usize,
+            params_s: params.clone(),
+            party_keys_s: party_keys_vec_l.clone(),
+            bc1_vec_s: bc1_vec_l.clone(),
```

Comment:
what's the difference between `_s` and `_l` ? 
originally `_l` suppose to mean "local" vector

Reply:
I used _s for structure members.. _l for local variable names same as structure member names